packages/core/core/src/migrations/database/5.0.0-discard-drafts.ts

```diff
@@ -57,16 +59,31 @@ const migrateUp = async (trx: Knex, db: Database) => {
 
     const uid = meta.uid as UID.ContentType;
     const model = strapi.getModel(uid);
+
     const hasDP = contentTypes.hasDraftAndPublish(model);
+    const isLocalized = strapi
+      .plugin('i18n')
+      .service('content-types')
+      .isLocalizedContentType(model);
+
     if (!hasDP) {
       continue;
     }
 
-    const discardDraft = async (entry: DocumentVersion) =>
+    const discardDraft = async (entry: DocumentVersion) => {
+      const params: DiscardDraftParams<typeof uid> = { documentId: entry.documentId };
+
+      // Only add the locale param if the model is localized
+      // ref. https://github.com/strapi/strapi/issues/20225
+      if (isLocalized) {
+        params.locale = entry.locale;
+      }
+
       strapi
         .documents(uid)
-        // Discard draft by referencing the documentId and locale
-        .discardDraft({ documentId: entry.documentId, locale: entry.locale });
+        // Discard draft by referencing the documentId (and locale if the model is localized)
+        .discardDraft(params);
```

Comment:
Shouldn't discardDraft support ignoring the locale instead 🤔  cc @Marc-Roig ?

Reply:
In the end, we'll probably have to fix it in both. I just saw [that](https://github.com/strapi/strapi/blob/v5/main/packages/core/core/src/services/document-service/repository.ts#L298-L299) in the method itself.

Reply:
(even though it appears the first methods are checking if the CT is localized)

Reply:
Yes, can you create an issue to also fix the fn 🙏

Reply:
True, the discardDraft should be smart enough to ignore it :/

Reply:
Actually, it seems like it is for the most part. There are checks for isLocalized inside each internationalization method used for the params.

The only bit I'm uncertain about is https://github.com/strapi/strapi/blob/v5/main/packages/core/core/src/services/document-service/repository.ts#L298-L299
Should we keep that as is? Are we assuming relations have locale in any case :thinking:

Reply:
Mmm we might need to not include locale if the relation target is not localized.
We are populating the target locale & document id fields, as that information is necessary in some scenarios (non localized CT -> localized CT)

We could manually ignore the "locale" field in that populate utility if the CT does not have I18n enabled. Wdyt?

Reply:
Yeah, indeed. In any case, we'll have to work a bit on those generated populate queries. We also have another issue where it doesn't respect morph relations and sends regular queries instead of fragments.

Reply:
Not sure it's the responsibility of the `getDeepPopulate` to sanitize the incoming selected fields though :thinking: 

I'll think a bit about it

Reply:
This is a problem we have faced in many other places, where we have to omit the "locale" or "publishedAt" field.
In any case, the document service assumes all content types include locale and publishedAt, even if those features are not enabled on the CT, we do include those columns.

Reply:
I believe this migration should include locale for all content type shttps://github.com/strapi/strapi/blob/eb316acc5c08cd34e14e22ce36c6e9806e701b70/packages/core/database/src/migrations/internal-migrations/5.0.0-03-locale.ts

shouldn't this fix the issue for v4->v5 migrations? 🤔

Reply:
Won't this be run before the one you've mentioned?

Reply:
Yeah, if we change the order of execution, the other migration might make this PR completely useless :slightly_smiling_face: